dspace-api/src/main/java/org/dspace/storage/rdbms/migration/V7_0_2020_10_31__CollectionCommunity_Metadata_Handle.java

```diff
@@ -0,0 +1,91 @@
+/**
+ * The contents of this file are subject to the license and copyright
+ * detailed in the LICENSE and NOTICE files at the root of the source
+ * tree and available online at
+ *
+ * http://www.dspace.org/license/
+ */
+package org.dspace.storage.rdbms.migration;
+
+import java.sql.PreparedStatement;
+import java.sql.SQLException;
+
+import org.dspace.handle.service.HandleService;
+import org.dspace.services.factory.DSpaceServicesFactory;
+import org.flywaydb.core.api.migration.BaseJavaMigration;
+import org.flywaydb.core.api.migration.Context;
+import org.springframework.dao.DataAccessException;
+import org.springframework.jdbc.core.JdbcTemplate;
+import org.springframework.jdbc.core.PreparedStatementSetter;
+import org.springframework.jdbc.datasource.SingleConnectionDataSource;
+
+/**
+ * Insert a 'dc.idendifier.uri' metadata record for each Community and Collection in the database.
+ * The value is calculated concatenating the canonicalPrefix extracted from the configuration
+ * (default is "http://hdl.handle.net/) and the object's handle suffix stored inside the handle table.
+ *
+ * @author Alessandro Martelli (alessandro.martelli at 4science.it)
+ */
+public class V7_0_2020_10_31__CollectionCommunity_Metadata_Handle extends BaseJavaMigration {
+    // Size of migration script run
+    protected Integer migration_file_size = -1;
+
+    @Override
+    public void migrate(Context context) throws Exception {
+
+        HandleService handleService = DSpaceServicesFactory
+                .getInstance().getServiceManager().getServicesByType(HandleService.class).get(0);
+
+        final String prefix = handleService.getCanonicalPrefix();
+
+        final String SQL_INSERT = "insert into metadatavalue "
```

Comment:
Have we fully tested this `SQL_INSERT` on Postgres *and* Oracle?  I'm a little nervous about hardcoding this SQL as a string in Java, simply because it means we cannot easily customize it for Postgres vs Oracle as necessary.

Normally, the process is to provide separate `*.sql` files under `dspace-api/src/main/resources/org/dspace/storage/rdbms/sqlmigration/[db-type]/`, rather than hardcoding SQL into Java.  

If this has *not* been tested on Oracle, I'd prefer we replace this logic with SQL files, even though I realize that'd require having two INSERT statements in each file (one for community & one for collection)

Reply:
We discussed a lot about it with @abollini. We option for a fully java based script since we have to make aware the database of the prefix value from configuration. This means we will still need a Java based script to properly initialize sql. Having said that in my opinion there is no problem in moving the sql part into separate files (and replicating it twice each).

Reply:
@alemarte : To clarify, my concern here is that *hardcoding SQL into Java* means that the SQL *must work on all database types*.  So, again, I'm wondering if this SQL has been tested on Oracle?  If we can prove it works for both Postgres & Oracle, then this approach is OK.  Otherwise, we *must* refactor this to use SQL scripts that are database type specific, as that allows us to customize the SQL per database type.

I'm a little nervous that this hardcoded SQL may not be cross-database compatible (as it's quite complex), but I hope I'm wrong.  In any case, I'd appreciate it if you could test this on Oracle to ensure it'll work there as well.  If so, then it's fine as-is.  If not, then we need to figure out a different solution that allows the Oracle SQL and Postgres SQL to be different.   One option might be to move to SQL files and use Flyway Placeholders (https://flywaydb.org/documentation/configuration/placeholder) in the SQL to represent the prefix -- but I'm not sure how easy that'd be, as we've not had to use placeholders yet in any other migration.

Reply:
@tdonohue I understand your concerns, unfortunately I can't say it works 100% on oracle. I've moved the insert parts into separate files. 
Now, the hard coded queries are listed below:

- [CREATE TABLE tmp_handleprefix (handleprefix text NOT NULL);](https://github.com/DSpace/DSpace/blob/65b69a2b29830c2f00a3dc6b8cd8bdca0faf4892/dspace-api/src/main/java/org/dspace/storage/rdbms/migration/V7_0_2020_10_31__CollectionCommunity_Metadata_Handle.java#L43)
- [INSERT INTO tmp_handleprefix (handleprefix) VALUES (?)](https://github.com/DSpace/DSpace/blob/65b69a2b29830c2f00a3dc6b8cd8bdca0faf4892/dspace-api/src/main/java/org/dspace/storage/rdbms/migration/V7_0_2020_10_31__CollectionCommunity_Metadata_Handle.java#L47)

Placeholders can't be used easily since we have no access to the Flyway instance within the migration script.